src/Features/Core/Portable/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeFromMembersCodeRefactoringProvider.cs

```diff
@@ -176,7 +140,10 @@ private bool HasOperator(INamedTypeSymbol containingType, string operatorName)
             if (equatableTypeOpt != null)
             {
                 constructedType = equatableTypeOpt.Construct(containingType);
-                return !containingType.AllInterfaces.Contains(constructedType);
+
+                // A ref struct can never implement an interface, therefore never add IEquatable to the selection
+                // options if the type is a ref struct.
```

Comment:
📝 It seems like it would be easier to read this method in the following form:

```
if (containingType.IsRefLikeType)
{
  // A ref struct can never implement an interface, therefore never add IEquatable to the selection
  // options if the type is a ref struct.
  constructedType = null;
  return false;
}

var equatableTypeOpt = semanticModel.Compilation.GetTypeByMetadataName(typeof(IEquatable<>).FullName);
if (equatableTypeOpt != null)
{
  constructedType = equatableTypeOpt.Construct(containingType);
  return !containingType.AllInterfaces.Contains(constructedType);
}

constructedType = null;
return false;
```

Reply:
I can see how this merge conflict resolution is the correct one (the PR originally implemented the check in the same form you used), but perhaps it could be simplified in a follow-up commit?